pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -434,6 +434,8 @@ public static class QueryOptionKey {
 
         /** Max number of groups GroupByDataTableReducer (running at broker) should return. */
         public static final String MIN_BROKER_GROUP_TRIM_SIZE = "minBrokerGroupTrimSize";
+        public static final String NUM_THREADS_FOR_FINAL_REDUCE = "numThreadsForFinalReduce";
```

Comment:
The naming is a little bit confusing. By reading the name, I'd imagine this is configuring the threads on broker reduce. This is parallelizing the final result extract for group by. We want to make it more specific

Reply:
updated.